extension/dynamicconfig/model/model.go

```diff
@@ -0,0 +1,47 @@
+// Copyright The OpenTelemetry Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+//
+// Contains common models for the dynamic config service. The corresponding
+// Proto() methods convert the model representation to a usable struct for
+// protobuf marshalling.
+
+package model
+
+import (
+	"fmt"
+	"hash"
+	"hash/fnv"
+)
+
+var hasher hash.Hash = fnv.New64a()
```

Comment:
Is the use of "hasher" thread safe?

Reply:
No, not inherently, I believe. A mutex was introduced to the backend to ensure its use is sync'ed.

The alternative would be to introduce syncing logic to the model definitions, but it seemed more elegant and concise to keep the model definitions simple, and move syncing to the backend.

Reply:
Can you pass the hasher as a parameter into combineHash()?

A not thread-safe global variable can cause quite some trouble if we are not careful. It is better to be defensive.